src/llvm/CompilerIntrinsics.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import {
+  Module,
+  Function as LLVMFunction,
+  FunctionType,
+  StructType,
+  Type as LLVMType,
+  LinkageTypes,
+  config,
+} from "llvm-node";
+import { llvmContext } from "./llvm-context.js";
+
+export class Intrinsics {
+  +_module: Module;
```

Comment:
I'm not sure what you mean. It just means that we  always have to initialize it with a Module in the constructor.

Reply:
You seem to be using a feature of Flow that is not documented in https://flow.org/en/docs/. My reading of that document is that + is a way to mark an interface property as covariant, which constrains the code accesses the underlying object via the interface to not write to that property, lest it inadvertently violate the type annotation of the property of the underlying object.

There is no documentation that I can find that informs me that + is a way to mark a class property as "must be initialized inside the constructor". I'm also much bemused by this interpretation since this should already the be case since _module MUST always have the type Module.

Where do I find more information on this use of +?

Reply:
https://flow.org/blog/2016/10/04/Property-Variance/

Reply:
I don't think a blog post from 2016 trumps the current documentation. Even so, I don't see anything there that suggests that + means that we always have to initialize the property.

Reply:
Sorry, I just meant that this is the only place where I know `+` was documented. Dunno if that changed, and haven't looked into how to use it myself.

Reply:
Covariant (read-only) properties are allowed to be initialized in class constructors in a recent-ish change to the type system. This preserves the semantics of covariant properties since you can’t assign back some larger type.

```js
type Y = {+p: number | string};

class X {
  +p: number;

  constructor() {
    this.p = 42; // Ok

    setTimeout(() => {
      this.p = 0; // Error
      (this: Y).p = 'foo'; // Error
    }, 0);
  }

  m() {
    this.p = 0; // Error
    (this: Y).p = 'foo'; // Error
  }
}
```

[try-Flow](https://flow.org/try/#0C4TwDgpgBAmlC8UDeBqMAuKA7ArgWwCMIAnKAHygGdhiBLLAcwF8BuAKDYGMAbAQ0spQAGsjZQoaTLkIl2YqJwD2WasRydgi4gAoAlKPHjgAC1qUAdGARQALACYWUAPROoAeQDWHQ1QjAAKrR4EIo4wNp6CAB8Bj5QJmaW1gAMji5QAKLExFryPtoJlJgwukmIAOQAZoqK5WmuWTnEeVBMADRQybrs4kzeUHiRSC2FZZ31mdm5+YXFpVYV1bUTjdOtbH1AA)